### PR TITLE
fix: Remove cassandra from log

### LINKED ISSUE #1452: Remove 'Cassandra' from Backend log
## Summary
Remove '**Cassandra**' from Backend log

## Motivation
We have many logs still using Cassandra
_Cassandra: Error setting_

_Cassandra sync write error, retrying_
Since we support both Cassandra / ScyllaDB, ScyllaDB user may find them misleading.

## Solution
Remove hardcoded _Cassandra_

### PR DESCRIPTION
**Pull request by cindyyan317**
Fix [#1452 ](https://github.com/XRPLF/clio/issues/1452)

### CODE REVIEW

File: src/data/BackendFactory.hpp [hunk 1]
```diff
@@ -51,8 +51,7 @@ make_Backend(util::Config const& config)
     auto const type = config.value<std::string>("database.type");
     std::shared_ptr<BackendInterface> backend = nullptr;
 
-    // TODO: retire `cassandra-new` by next release after 2.0
-    if (boost::iequals(type, "cassandra") or boost::iequals(type, "cassandra-new")) {
+    if (boost::iequals(type, "cassandra")) {
```
Comment (kuznetsss) [inline src/data/BackendFactory.hpp:54]: This is a breaking change for config. Are you sure it is ok to do that?

**Review by kuznetsss — APPROVED** [review]
LGTM, just one question.

File: src/data/BackendFactory.hpp (hunk 1 above)
Comment (cindyyan317) [reply src/data/BackendFactory.hpp:54]: 👍 Good call. 
I think "cassandra-new" is only used internally, when we compared the performance between scylladb and cassandra. Will check.

Comment (cindyyan317) [reply src/data/BackendFactory.hpp:54]: Checked with @godexsoft . It is never announced, should be safe to delete.

**Pull request merged**